backend/graminsta/core/services.py

```diff
@@ -33,3 +35,34 @@ def create_userinfo(validated_data):
         gender=validated_data.pop('gender')
     )
     return user_info
+
+
+def user_authentication(request):
```

Comment:
Usually, the name of the method should be some verb + noun, in your case, I think you can just use `create_authentication_token`, since `request` is a concept between client-server, you don't need to know that in your service APIs, you can pass in the `username` and `password`, return a token.